src/java/voldemort/server/VoldemortConfig.java

```diff
@@ -880,7 +880,7 @@ private void initializeStateFromProps() throws UndefinedPropertyException {
         this.readOnlyMaxValueBufferAllocationSize = this.allProps.getInt(READONLY_MAX_VALUE_BUFFER_ALLOCATION_SIZE);
         this.readOnlyCompressionCodec = this.allProps.getString(READONLY_COMPRESSION_CODEC);
         this.modifiedProtocol = this.allProps.getString(READONLY_MODIFY_PROTOCOL);
-        this.modifiedPort = this.allProps.getInt(READONLY_MODIFY_PORT);
+        this.modifiedPort = this.allProps.getNullableInt(READONLY_MODIFY_PORT);
```

Comment:
It's implied that it returns an `Integer`, since a primitive `int` cannot be null. Anyhow, I've renamed the function to `getNullableInteger` as you suggest.